pandas/__init__.py

```diff
@@ -4,17 +4,13 @@
 __docformat__ = 'restructuredtext'
 
 try:
-    from . import hashtable, tslib, lib
-except Exception:  # pragma: no cover
-    import sys
-    e = sys.exc_info()[1]  # Py25 and Py3 current exception syntax conflict
-    print(e)
-    if 'No module named lib' in str(e):
-        raise ImportError('C extensions not built: if you installed already '
-                          'verify that you are not importing from the source '
-                          'directory')
-    else:
-        raise
+    from pandas import hashtable, tslib, lib
+except ImportError as e:  # pragma: no cover
+    module = str(e).lstrip('cannot import name ')  # hack but overkill to use re
```

Comment:
if you mean the part about finding out which module causes the `ImportError`, yes it's not really necessary. But I suppose it was done to make the error message even more specific, i.e. "can't import hashtable, you need to build C extensions" vs "you need to build C extensions"

both would be much better than the current error that says "can't import hashtable" http://stackoverflow.com/questions/14422976/importing-pandas-shows-importerror-cannot-import-name-hashtable